[Peter Fung] iP

fungusta opened the pull request:
#  💯 **Duke v0.1** 💯 

Individual Project for _CS2103T_

## What 🤔 

An interactive chatbot that helps users to keep track of tasks

## Features 😮 

- Add Tasks
- Mark Tasks as done or undone
- List out all tasks
- Differentiate different tasks
- Delete Tasks off list
- Find Tasks in the list
- Save Tasks in a file on your computer
- Prompts when unknown command is input
- Exit

## Commands that can be run 👨‍💻 

- todo
- deadline
- event
- delete
- mark
- list
- find
- bye

**Some examples of command formats** 💬 
```
todo read book
deadline homework /by 27/06/2023 1800
event mother's graduation /at 03/06/9009 1369 
delete 2
mark 1
list
find book
bye
```


## Additional Coding Implementations ➕ 

1. Text User Interface Testing
2. Enumerations
3. More OOP
4. Packages
5. Gradle
6. Junit
7. Jar
8. Java Documentation
9. Coding Standard

## Levels Completed 🆙 

- [x] Level 1
- [x] Level 2
- [x] Level 3
- [x] Level 4
- [x] Level 5
- [x] Level 6
- [x] Level 6
- [x] Level 7
- [x] Level 8
- [x] Level 9
- [x] Level 10

## Additional Information 🔗 

View the GitHub repository [here](https://github.com/fungusta/ip)

**Quote I ~~Love~~ Live By**
> " I'm Batman." - Bruce Wayne/Batman, Batman Begins.

The main method:
```
public static void main(String[] args) {
    new Duke("./data", "./data/Duke.txt").run();
}
```

Commit:
Add Gradle support

fungusta committed:
Increment Level-1

fungusta committed:
Increment Level 2

fungusta committed:
Removed commented out code

fungusta committed:
Increment Level 3

fungusta committed:
Increament Level 4

fungusta committed:
Corrected Java Version to Version 11

fungusta committed:
Increment  A-TextUiTesting

fungusta committed:
Increment Level-5

fungusta committed:
Increment Level-6

fungusta committed:
Made the code neater

fungusta committed:
Increament A-Enums

fungusta committed:
Add save feature to Duke

fungusta committed:
Update Deadlines and Events to contain DateTime instead of String of dates

fungusta committed:
Merge branch 'branch-Level-7'

fungusta committed:
Update method names to follow code style

fungusta committed:
Merge branch 'branch-Level-8'

fungusta committed:
Create OOP files

fungusta committed:
Merge branch 'master' of https://github.com/fungusta/iP

fungusta committed:
Make the code more OOP

fungusta committed:
Remove Duplicate File

fungusta committed:
Update .gitignore

fungusta committed:
Increment A-Package

fungusta committed:
Merge remote-tracking branch 'origin/add-gradle-support'

fungusta committed:
Add A-Gradle and A-JUnit

fungusta committed:
Update .gitignore

fungusta committed:
Added JavaDocs

fungusta committed:
Adapt code to follow coding standard

fungusta committed:
Added Find function for Duke

fungusta committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/task/Deadlines.java
#	src/main/java/duke/task/Events.java
#	src/main/java/duke/task/ToDos.java

fungusta committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/UI.java
#	src/main/java/duke/task/Task.java

fungusta committed:
Rename Ui class to follow code style

fungusta committed:
Merge branch 'master' of https://github.com/fungusta/ip

fungusta committed:
fix: correct code to follow checkstyle

parnikkapore commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+    private final Storage storage;
+    private final TaskList taskList;
+    private final Ui ui;
+    private boolean isBye; //Indicates if bye was input by user
```
```suggestion
    private boolean isBye; // Indicates if bye was input by user
```

Should there be a space between the `//` and the comment? I think the space is required in many coding standards, although not specifically CS2103T's.

This also applies to more inline comments below.

parnikkapore commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+
+    /**
+     * Runs the Duke program.
+     * <p>
```
```suggestion
     * 
```

Should this line be blank? As far as I can tell, this is required by the [coding standard](https://se-education.org/guides/conventions/java/index.html#comments).

parnikkapore commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,130 @@
+    public static Command parseInput(String input) throws DukeException {
+        if (!input.contains(" ")) {
+            switch (Instructions.valueOf(input)) {
+            case bye:
```
I like how you indented the case labels correctly 🙂

parnikkapore commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,49 @@
+     * @return arraylist of tasks that contains keyword.
+     */
+    public TaskList findKeyWord(String keyword) {
+        TaskList findList = new TaskList();
```
Should `findList` be named `tasksFound` instead, so that the name has a plural form (see the [coding standard](https://se-education.org/guides/conventions/java/index.html#naming))?

Personally I would've considered this passable as there is a `-List` suffix indicating the collectionness.

parnikkapore commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,62 @@
+/**
+ * Represents an event task.
+ */
+public class Events extends Task {
```
Should this class be named "Event"? It represents a single event and not a collection of them.

parnikkapore commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,29 @@
+public class ParserTest {
+    @Test
+    @DisplayName("Check parse input for bye and list")
+    public void checkParseInputByeAndList() {
```
Should this be named according to the three-part format (perhaps `parseInput_bye_list_returnsSingleton`?) This also applies to the other tests.

parnikkapore reviewed (changes requested):
Nice job on following the indentation and braces placement guidelines. There are a couple of naming issues though, which should be a pretty quick fix. Waiting on code quality review.

parnikkapore reviewed (commented):
Nice job on following the indentation and braces placement guidelines. There are a couple of naming issues though, which should be a pretty quick fix. Waiting on code quality review.

fungusta committed:
chore: added JavaFX files

fungusta committed:
feat: GUI for Duke

fungusta committed:
Add assert statements to come classes

There are a few classes that require certain variables not be negative

Including assertions would allow that the variables are never that value at the important times

fungusta committed:
Update files to upkeep code quality

As there were several methods with more the 30 LOC which had to be refactored

This was a big step in achieving proper code quality.

fungusta committed:
Merge pull request #2 from fungusta/branch-A-Assertions

Add assert statements to come classes

fungusta committed:
Merge branch 'master' into branch-A-CodeQuality

fungusta committed:
Merge pull request #1 from fungusta/branch-A-CodeQuality

Update files to upkeep code quality

fungusta committed:
Set up Continuous Integration

fungusta committed:
Update gradle.yml file

fungusta committed:
Revert "Update gradle.yml file"

This reverts commit 323992af9916f5a63c6500dad12a28bd93fb873f.

fungusta committed:
Revert "Set up Continuous Integration"

This reverts commit d175b13979f5a34b430730327c9e62cf5f3a2a7f.

fungusta committed:
Feat: Add Client functionalities with add and delete commands

fungusta committed:
feat: Add Save and List function to client list

fungusta committed:
feature: add javadocs

fungusta committed:
feat: Add User Guide

fungusta committed:
feat: Updated User Guide

fungusta committed:
fix: Update code quality and style

fungusta committed:
fix: Fix some bugs with save file
